### PR TITLE
Ndm/price service

### PR DESCRIPTION
**Pull request by kasiaturbiarz**

### CODE REVIEW

File: src/Nethermind/Nethermind.DataMarketplace.Core/Services/GasPriceService.cs [hunk 1]
```diff
@@ -158,6 +157,12 @@ public async Task SetPaymentClaimGasPriceAsync(UInt256 gasPrice)
 
         public async Task UpdateGasPriceAsync()
         {
+            var currentTime = _timestamper.UnixTime.Seconds;
+            if (_updatedAt + _updateInterval > currentTime || Interlocked.Exchange(ref _updatedAt, currentTime) == currentTime)
+            {
+                return;
+            }
+
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.DataMarketplace.Core/Services/GasPriceService.cs:165]: This looks very odd to me. I am not sure it works as intended.

File: src/Nethermind/Nethermind.DataMarketplace.Core/Services/PriceService.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            _updateInterval = updateInterval;
+        }
+        
+        public async Task UpdateAsync(params string[] currencies)
+        {
+            var currentTime = _timestamper.UnixTime.Seconds;
+            if (_updatedAt + _updateInterval > currentTime || Interlocked.Exchange(ref _updatedAt, currentTime) == currentTime)
+            {
+                return;
+            }
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.DataMarketplace.Core/Services/PriceService.cs:43]: Again I don't think its correct and could use a test

**Review by LukaszRozmej — CHANGES_REQUESTED** [review]

File: src/Nethermind/Nethermind.DataMarketplace.Core/Services/GasPriceService.cs (hunk 1 above)
Comment (tkstanczak) [reply src/Nethermind/Nethermind.DataMarketplace.Core/Services/GasPriceService.cs:165]: != current time?

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.DataMarketplace.Core/Services/GasPriceService.cs (hunk 1 above)
Comment (kasiaturbiarz) [reply src/Nethermind/Nethermind.DataMarketplace.Core/Services/GasPriceService.cs]: I inverted IF and I changed this condition

**Pull request merged**